ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/definition/package.scala

```diff
@@ -64,10 +65,11 @@ package object definition {
   import pl.touk.nussknacker.engine.graph.NodeDataCodec._
   object ComponentTemplate {
     def create(`type`: ComponentType, node: NodeData, categories: List[String], branchParametersTemplate: List[evaluatedparam.Parameter] = List.empty): ComponentTemplate =
-      ComponentTemplate(`type`, `type`.toString, node, categories, branchParametersTemplate)
+      ComponentTemplate(`type`, `type`.toString, node, categories, branchParametersTemplate, componentId = None)
   }
 
-  @JsonCodec(encodeOnly = true) case class ComponentTemplate(`type`: ComponentType, label: String, node: NodeData, categories: List[String], branchParametersTemplate: List[evaluatedparam.Parameter] = List.empty)
+  //TODO: componentId is temporary field and it is work around for components duplication across multiple scenario types
+  @JsonCodec(encodeOnly = true) case class ComponentTemplate(`type`: ComponentType, label: String, node: NodeData, categories: List[String], branchParametersTemplate: List[evaluatedparam.Parameter] = List.empty, componentId: Option[ComponentId])
```

Comment:
Why `componentId` is needed in `ComponentTemplate`? Template is for creating node based on component and should has similar fields to be used in `Node`.  During writting it, I've realized that it is not a `Component` `Template`. It is `NodeTemplate` based on `Component`

Reply:
Let's look at DefaultComponentService 78 line.  To fetch all components we based on `UIProcessObjectsFactory.prepareUIProcessObjects` which is preparing ComponentTempalte - we use this DTO to create our ComponentListElement.

But right now I see we can use `componentsConfig`, because we put here merged componentsConfig (fixed - config, dynamic - from ProcessCreator). WDT?